### PR TITLE
feat(translation): Does() family patterns — Phase 1 (status, negation, combat, possession, motion)

### LINKED ISSUE #61: feat(logic): Does() translation strategy — 289 VerbComposition sites
## Summary
Translate Does() VerbComposition sites using **family-based sink translation** (B') with **limited per-site patches** (C) for high-density clusters.

## Strategy Decision (2026-03-22)
Consulted Codex, ChatGPT Thinking, and ChatGPT Pro. **All three agreed:**

- **A (patch Does() itself): REJECTED** — Does() returns subject+verb only; callers concatenate English tails via `+`, producing mixed-language output
- **B' (family-based sink translation): PRIMARY** — use existing MessageLogPatch / PopupTranslationPatch as sink, add family translators (not per-site regex)
- **C (per-site Harmony patches): LIMITED** — only for high-density clusters where sink family matching is fragile

## Evidence (from decompiled source analysis)
- 307 deduplicated call sites across 136 files (71 files are singletons)
- 296/307 sites use `Does(...) + English tail` pattern
- HandleMessage does NOT use Does() — completely separate code path
- Hot clusters: MissileWeapon (27), ITeleporter (8), TradeUI (8), Combat (7), Flight (5)

## Implementation Plan
1. Cluster analysis: group 307 sites by `verb + tail shape` into families
2. Implement family translators in MessagePatternTranslator (sink B')
3. Per-site patches for MissileWeapon, Combat, TradeUI, ITeleporter, Flight clusters (limited C)
4. Singleton sites: best-effort via sink family matching

## Prerequisites
- Existing MessageLogPatch / PopupTranslationPatch sink infrastructure ✅
- MessagePatternTranslator missing-pattern observation ✅

## Acceptance Criteria
- [ ] Family clustering analysis complete (target: 307 → 40-100 families)
- [ ] Top 6 clusters implemented as family translators
- [ ] High-density clusters (MissileWeapon, Combat, TradeUI) have per-site patches
- [ ] Tests cover new families

### PR DESCRIPTION
**Pull request by ToaruPen**
## Summary
Phase 1 of #61: add family-based sink translation coverage for Does() messages and L2 integration coverage for status predicate routing.

### Families covered
| Family | Coverage |
|--------|----------|
| Status Predicate | Added family patterns and L1 coverage |
| Negation/Lack | Added family patterns and L1 coverage |
| Combat (3rd person) | Added family patterns and L1 coverage |
| Possession/Lack | Added family patterns and L1 coverage |
| Motion/Direction | Added family patterns and L1 coverage |

### Additional validation
- Added L2 `MessageLogPatch` coverage for a status predicate translation path
- Preserved color handling through the existing strip/restore pipeline
- Verified the full C# build and test suite after the final test addition

### Validation
- [x] `dotnet build Mods/QudJP/Assemblies/QudJP.csproj`
- [x] `dotnet test Mods/QudJP/Assemblies/QudJP.Tests/QudJP.Tests.csproj`

Closes #61

## Summary by CodeRabbit

* **Tests**
  * 動詞類ごとの翻訳パターン（色表現やラッパーを含む複数表現）を検証する新規テストフィクスチャを追加しました。
  * メッセージログの翻訳挙動を確認するパッチ適用テストを追加しました。
  * パターン未一致・空文字・null・直接翻訳マーカーの扱い等の境界ケースを追加で検証します。

* **Chores**
  * 日本語ローカライズのパターンを大幅に拡張し、追加ステータス効果、戦闘・ダメージ表現、移動・行動、各種ユーティリティ系メッセージ（所持・資源・視認・聞こえ等）をサポートしました。

### CODE REVIEW

**Comment by ToaruPen** [comment]
@coderabbitai review

File: Mods/QudJP/Assemblies/QudJP.Tests/L1/DoesVerbFamilyTests.cs [hunk 1]
```diff
@@ -0,0 +1,176 @@
...
+                AppDomain.CurrentDomain.BaseDirectory,
+                "..",
+                "..",
+                "..",
+                "..",
+                "..",
+                "Localization",
+                "Dictionaries",
+                "messages.ja.json"),
+            patternFilePath);
```
[the remark this answers is not in the document]
Comment (ToaruPen) [reply Mods/QudJP/Assemblies/QudJP.Tests/L1/DoesVerbFamilyTests.cs]: Acknowledged — the relative path is fragile but matches the existing test pattern used throughout the project (MessagePatternTranslatorTests, XDidYTranslationPatchTests, etc.). Changing it would require a project-wide refactor. Leaving as-is for consistency.

File: Mods/QudJP/Localization/Dictionaries/messages.ja.json [hunk 2]
```diff
@@ -217,10 +325,178 @@
...
+      "template": "{t0}は{2}するのに十分な{1}がない！"
+    },
+    {
+      "pattern": "^(?:The |the |[Aa]n? )?(.+?) (?:doesn't|don't) have enough (.+?) to (.+?)\\.$",
+      "template": "{t0}は{2}するのに十分な{1}がない。"
+    },
+    {
+      "pattern": "^(?:The |the |[Aa]n? )?(.+?) (?:doesn't|don't) have enough (.+?) to (.+?)$",
+      "template": "{t0}は{2}するのに十分な{1}がない"
+    },
```
[the remark this answers is not in the document]
Comment (ToaruPen) [reply Mods/QudJP/Localization/Dictionaries/messages.ja.json]: Fixed in e69c36d. All three have-enough patterns now use `{t0}`, `{t1}`, `{t2}` so resource names and action names pass through `Translator.Translate`.

File: Mods/QudJP/Assemblies/QudJP.Tests/L1/DoesVerbFamilyTests.cs [hunk 3]
```diff
@@ -0,0 +1,131 @@
...
+    // Plain text
+    [TestCase("The 熊 falls to the ground.", "熊は地面に倒れた。")]
+    [TestCase("You fall to the ground.", "あなたは地面に倒れた。")]
+    [TestCase("The スナップジョー falls asleep.", "スナップジョーは眠りに落ちた。")]
+    // Color-wrapped (ConsequentialColor)
+    [TestCase("{{g|The 熊 falls to the ground.}}", "{{g|熊は地面に倒れた。}}")]
+    public void Translate_MotionDirectionFamily(string input, string expected)
+    {
+        AssertTranslated(input, expected);
+    }
```
[the remark this answers is not in the document]
Comment (ToaruPen) [reply Mods/QudJP/Assemblies/QudJP.Tests/L1/DoesVerbFamilyTests.cs]: Fixed in 34a71b8. Added 4 edge-case tests shared across all families:
- `Translate_ReturnsOriginal_WhenNoPatternMatches` (fallback to English)
- `Translate_ReturnsEmpty_WhenInputIsEmpty`
- `Translate_ReturnsEmpty_WhenInputIsNull` (Translate returns empty for null per implementation)
- `Translate_SkipsTranslation_WhenDirectTranslationMarkerPresent` (\x01 marker passthrough)

571/571 tests pass.

**Comment by ToaruPen** [comment]
@coderabbitai review
